### DISCUSSION TITLE
Does Ultralytics HUB supports uploading a trained yolo model file ?

### DISCUSSION THREAD

**Discussion by singyiu** [ultralytics/hub#501] [Q&A]
Does Ultralytics HUB supports uploading a trained yolo model file ?
Such that we don't have to upload the whole dataset and re-train it.
Thanks team !

**Comment by nolby003** [comment]
Not only you cannot upload a trained model, the YOLO can only detect images based on the images it has already trained on, I offer it some image off the internet and it either false detects it as something else or nothing at all - I have 26 classes and it is frustrating how much time I have wasted on this CNN that does not work, I think resnet or VGG16 will be my next attempt.
Btw the app thinks everything is a tomato leaf no matter what, even a bounding box on the side of my camera window, wa?

**Reply by mirage-4000** [reply]
for me it work really well even with web image wich the model never see, I only have a bug when using the app on a newer phone , I also have a detection on the top of my screen for no reasons but if your model don´t recognize image of the web it's maybe cause your dataset is not diverse enough.

**Reply by throughthetrees** [reply]
How is this not a thing?

I have a YOLO I trained in colab using ultralytics...

and I can't export it to hub? 

I want to demo it on the ultralytics app on a colleagues iPhone... and I cant?

**Reply by sergiuwaxmann** [reply]
@throughthetrees The upload of trained models feature will be implemented most likely later this year. Apologies for the inconvenience.

**Reply by throughthetrees** [reply]
@sergiuwaxmann Excited to hear that.
